pytition/petition/views.py

```diff
@@ -1526,4 +1526,81 @@ def image_upload(request):
     name = storage._save(path, file)
     newrelpath = os.path.relpath(name, storage.location)
 
-    return JsonResponse({'location': storage.base_url + newrelpath})
\ No newline at end of file
+    return JsonResponse({'location': storage.base_url + newrelpath})
+
+# /transfer_petition/<int:petition_id>
+# Transfer a petition to another org or user
+@login_required
+def transfer_petition(request, petition_id):
+    pytitionuser = get_session_user(request)
+    try:
+        petition = petition_from_id(petition_id)
+    except:
+        messages.error(request, _("This petition does not exist."))
+        return redirect("user_dashboard")
+
+    ctx = {'user': pytitionuser,
+           'petition': petition}
+
+    if petition.owner_type == "org":
+        org = petition.owner
+        if not org.is_allowed_to(pytitionuser, "can_modify_permissions"):
+            messages.error(request, _("You don't have the permission to transfer a petition from Organization '{}'"
+                                      .format(petition.owner)))
+            return redirect("org_dashboard", petition.owner)
+
+    if request.method == "GET":
+        return render(request, "petition/transfer_petition.html", ctx)
+
+    if request.method == "POST":
+        owner_type = request.POST.get('new_owner_type', '')
+        new_owner_name = request.POST.get('new_owner_name', '')
+        org = None
+        user = None
+        notFound = False
+        if owner_type == '' or new_owner_name == '':
+            messages.error(request, _("Incorrect request, something went wrong."))
+            return redirect(request, "petition/transfer_petition.html", ctx)
+        if owner_type == "org":
+            try:
+                org = Organization.objects.get(slugname=new_owner_name)
+            except:
+                notFound = True
+        if owner_type == "user":
+            try:
+                user = PytitionUser.objects.get(user__username=new_owner_name)
+            except:
+                notFound = True
+
+        if notFound:
+            messages.error(request, _("You tried to transfer a petition to a non-exiting account"))
+            return redirect("transfer_petition", petition.id)
+
+        try:
+            petition.transfer_to(user, org)
+            messages.success(request, _("Petition successfully transfered!"))
+            return redirect("user_dashboard")
+        except ValueError:
+            messages.error(request, _("Something went wrong while transferring this petition."))
+            return redirect("transfer_petition", petition.id)
+
+
+@login_required
+def search_users_and_orgs(request):
+    q = request.GET.get('q', '')
```

Comment:
I don't know if `q` is very explicit (both for the variable name and the query parameter) :)

Reply:
let's name it `query` then?

Reply:
Sure, I think that's a bit clearer :)